src/public/modules/forms/admin/controllers/admin-form.client.controller.js

```diff
@@ -179,13 +181,39 @@ function AdminFormController(
    * @returns Promise
    */
   $scope.updateForm = (update) => {
-    return FormApi.update({ formId: $scope.myform._id }, { form: update })
-      .$promise.then((savedForm) => {
-        // Updating this form updates lastModified
-        // and also updates myform if a formToUse is passed in
-        $scope.myform = savedForm
-      })
-      .catch(handleUpdateError)
+    const updateType = get(update, 'type')
+
+    switch (updateType) {
+      case UPDATE_FORM_TYPES.UpdateField: {
+        const { fieldId, body } = update
+        return $q
+          .when(
+            AdminFormService.updateSingleFormField(
+              $scope.myform._id,
+              fieldId,
+              body,
+            ),
+          )
+          .then((updatedFormField) => {
+            // merge back into the form fields
+            const updateIndex = $scope.myform.form_fields.findIndex(
+              (f) => f._id === fieldId,
+            )
+            if (updateIndex !== -1) {
```

Comment:
should we throw an error here if `updateIndex === -1`?

Reply:
a58c9d2